### PR TITLE
Add Kyber swap aggregator

### LINKED ISSUE #1969: [ADD] Implement new swap aggregator: Kyber Swap
## Overview
Integrate KyberSwap as an additional swap aggregator to access better rates and broader token support while collecting protocol fees (50 bps). This implementation will follow KyberSwap's Aggregator API specifications to enable seamless token swaps with built-in fee collection, if any details are missing, please refer to the KyberSwap documentation.

### 1. Setup & Configuration
- [ ] Add KyberSwap API base URL: `https://aggregator-api.kyberswap.com`
- [ ] Configure platform-specific client ID headers, **important to include this ID otherwise we will get rate limited**:
  - iOS: `X-Client-Id: vultisig-ios`
  - Android: `X-Client-Id: vultisig-android`
  - Extension: `X-Client-Id: vultisig-v0`

### 2. Route Discovery (`GET /{chain}/api/v1/routes`)
- [ ] Required query parameters:
  - `tokenIn`: Source token address (checksummed)
  - `tokenOut`: Destination token address (checksummed)
  - `amountIn`: Amount in smallest unit (string)
  - `saveGas`: true (for gas optimization)
  - `gasInclude`: true (to include gas in the swap)
  - `slippageTolerance`: 100 (1.0%)

### 3. Transaction Building (`POST /{chain}/api/v1/route/build`)
- [ ] Required request body:
  ```json
  {
    "routeSummary": {
      // From route discovery response
    },
    "slippageTolerance": 100,
    "sender": "0xUserAddress",
    "recipient": "0xUserAddress",
    "skipSimulateTx": false,
    "source": "",
    "enableSlippageProtection": true,
    "feeReceiver": "0xa4a4f610e89488eb4ecc6c63069f241a54485269",
    "feeAmount": "50",
    "chargeFeeBy": "currency_out",
    "isInBps": true
  }
  ```
- [ ] Handle response fields:
  - `data.routerAddress`: Router contract to interact with
  - `data.amountIn`: Input amount
  - `data.amountInUsd`: USD value of input
  - `data.amountOut`: Expected output amount
  - `data.amountOutUsd`: USD value of output
  - `data.gas`: Estimated gas cost
  - `data.gasUsd`: USD value of gas
  - `data.data`: Encoded transaction data

### 4. Transaction Execution
- [ ] Get user approval for swap details
- [ ] Submit transaction using wallet's Web3 provider
- [ ] Monitor transaction status and confirmations
- [ ] Handle transaction replacement (speed up/cancel)

### 5. Error Handling
- [ ] HTTP 400: Invalid parameters (validate inputs)
- [ ] Transaction failures:
  - Reverted transactions
  - Slippage exceeded
  - Insufficient gas
  - Expired quotes (quotes are valid for 60s)

### 6. Testing
- [ ] Test with native token (ETH/BNB/etc.) and ERC20 tokens
- [ ] Verify fee collection (0.5% of output)
- [ ] Test edge cases:
  - Small/large amounts
  - Low liquidity pairs
  - Network congestion scenarios
  - Expired quotes

## Security Considerations
- [ ] Validate all API responses
- [ ] Use HTTPS for all requests
- [ ] Implement proper error boundaries
- [ ] Log errors for debugging (excluding sensitive data)

## References
- [KyberSwap Aggregator API](https://docs.kyberswap.com/kyberswap-solutions/kyberswap-aggregator/aggregator-api-specification/evm-swaps)
- [Build Transaction Endpoint](https://docs.kyberswap.com/kyberswap-solutions/kyberswap-aggregator/aggregator-api-specification/evm-swaps#post-chain-api-v1-route-build)

### PR DESCRIPTION
**Pull request by aminsato**
## Description

Please include a summary of the change and which issue is fixed. 

Fixes #1969

## Which feature is affected?
- [ ] Create vault ( Secure / Fast) - Please ensure you created a Secure vault & fast vault
- [ ] Sending  - Please attach a tx link here
- [ ] Swap - Please attach a tx link for the swap here
- [ ] New Chain / Chain related feature  -  Please attach tx link here

## Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

## Additional context

### CODE REVIEW

File: data/src/main/kotlin/com/vultisig/wallet/data/mappers/KeysignPayloadProtoMapper.kt [hunk 1]
```diff
@@ -77,6 +83,42 @@ internal class KeysignPayloadProtoMapperImpl @Inject constructor() : KeysignPayl
...
+                    SwapPayload.Kyber(
+                        KyberSwapPayloadJson(
+                            fromCoin = requireNotNull(it.fromCoin).toCoin(),
+                            toCoin = requireNotNull(it.toCoin).toCoin(),
+                            fromAmount = BigInteger(it.fromAmount),
+                            toAmountDecimal = BigDecimal(it.toAmountDecimal),
+                            quote = requireNotNull(it.quote).let { it ->
+                                KyberSwapQuoteResponse(
+                                    code = 0,
+                                    message = "Success",
```
[the remark this answers is not in the document]
Comment (aminsato) [reply data/src/main/kotlin/com/vultisig/wallet/data/mappers/KeysignPayloadProtoMapper.kt:93]: https://github.com/vultisig/vultisig-ios/blob/7230ce776185fc52f3859461fd5f903db983ee6a/VultisigApp/VultisigApp/Services/Keysign/KeysignMessage%2BProtoMappable.swift#L178C2-L179C1

File: data/src/main/kotlin/com/vultisig/wallet/data/api/models/quotes/KyberSwapQuoteJson.kt.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            data = data.copy(amountOut = value)
+        }
+
+    fun gasForChain(chain: Chain): Long {
+        val baseGas = data.gas?.toLongOrNull() ?: 600_000L
+        val gasMultiplierTimes10 = when (chain) {
+            Chain.Ethereum -> 14L
+            Chain.Arbitrum, Chain.Optimism, Chain.Base, Chain.Polygon, Chain.Avalanche, Chain.BscChain -> 20L
+            else -> 16L
+        }
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/models/quotes/KyberSwapQuoteJson.kt.kt:39]: why this values?

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            url {
+                path(
+                    chain.raw.lowercase(),
+                    "api/v1/routes",
+                )
+                parameters.append(
+                    "tokenIn",
+                    sourceAddress
+                )
+                parameters.append(
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:61]: Please group the with `apply`
url {
parameters.apply {
append("tokenIn", sourceAddress)
append("tokenOut", destinationAddress)
append("amountIn", amount)
append("saveGas", "false")
append("gasInclude", "true")
append("slippageTolerance", SLIPPAGE_TOLERANCE.toString())
append("isAffiliate", isAffiliate.toString())
parameter("sourceIdentifier", if (isAffiliate) CLIENT_ID else null)
parameter("referrerAddress", if (isAffiliate) REFERRER_ADDRESS else null)
}

File: data/src/main/kotlin/com/vultisig/wallet/data/api/models/KyberJson.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import kotlinx.serialization.json.JsonElement
+
+sealed class KyberSwapError(message: String) : Exception(message) {
+    class ApiError(val code: Int, message: String, val details: List<String>?) :
+        KyberSwapError(message)
+
+    class TransactionWillRevert(message: String) : KyberSwapError(message)
+    class InsufficientAllowance(message: String) : KyberSwapError(message)
+    class InsufficientFunds(message: String) : KyberSwapError(message)
+}
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/models/KyberJson.kt:15]: these types are not used anywhere. Was the purpose to have specific UI? isn't just generic exceptio with message enough?

Can we just use bodyOrThrow() which will contains the message?

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+            )
+        } catch (e: KyberSwapError.TransactionWillRevert) {
+            if (e.message?.contains("TransferHelper") == true) {
+                buildTransaction(
+                    chain,
+                    routeResponse,
+                    from,
+                    enableGasEstimation = false,
+                    isAffiliate
+                )
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:155]: Why returning transaction if it will revert? user will just burnt fees

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberSwap.kt [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            keysignPayload.coin.coinType,
+            vaultHexPublicKey,
+            vaultHexChainCode
+        ).getSignedTransaction(
+            inputData,
+            signatures
+        )
+    }
+
+    fun getPreSignedApproveInputData(
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberSwap.kt:63]: This is not used

File: data/src/main/kotlin/com/vultisig/wallet/data/common/ByteArrayExtensions.kt [hunk 7]
```diff
@@ -10,4 +11,8 @@ fun ByteArray.toKeccak256(): String {
 fun ByteArray.toKeccak256ByteArray(): ByteArray {
     val digest = Keccak.Digest256()
     return digest.digest(this)
+}
+
+fun  ByteArray.toByteString(): ByteString {
+    return ByteString.copyFrom(this)
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/common/ByteArrayExtensions.kt:17]: Do we really need it? Can we reuse the one from `ByteString.kt`

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberSwap.kt [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                            quote.tx.value.toBigInteger().toByteArray()
+                                ?: BigInteger.ZERO.toByteArray()
+                        )
+                    ).setData(quote.tx.data.removePrefix("0x").toByteStringOrHex())
+                ).build()
+            )
+        var gasPrice = quote.tx.gasPrice.toBigIntegerOrNull() ?: BigInteger.ZERO
+        if (keysignPayload.coin.chain == Chain.Arbitrum) {
+            // set gasPrice to null/zero for envelope transaction on arbitrum chain
+            gasPrice = BigInteger.ZERO
```
[the remark this answers is not in the document]
Comment (JaimeToca) [reply data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberSwap.kt:104]: (For example, code from 1inch is likely repeated here)

Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberSwap.kt:104]: why gas price only 0 in arbitrum?

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+        chain: Chain, routeResponse: KyberSwapRouteResponse, from: String,
+        enableGasEstimation: Boolean, isAffiliate: Boolean
+    ): KyberSwapQuoteJson {
+
+
+        val buildPayload = KyberSwapBuildRequest(
+            routeSummary = routeResponse.data.routeSummary,
+            sender = from,
+            recipient = from,
+            slippageTolerance = 100,
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:172]: slippageTolerance = 100 is actually quite dangerous. Is it used anywhere? Default value should be of 2.5-3% max

File: data/src/main/kotlin/com/vultisig/wallet/data/utils/Serializer.kt [hunk 10]
```diff
@@ -145,6 +147,29 @@ class LiFiSwapQuoteResponseSerializerImpl @Inject constructor(private val json:
...
+                json.decodeFromJsonElement<KyberSwapQuoteJson>(jsonObject)
+            )
+        } else {
+            KyberSwapQuoteDeserialized.Error(
+                json.decodeFromJsonElement<String>(jsonObject)
+            )
+        }
+    }
+}
+
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/utils/Serializer.kt:172]: Do we really need this?

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+
+        val route = json.decodeFromString<KyberSwapRouteResponse>(responseString)
+        return KyberSwapQuoteDeserialized.Result(
+            buildTransactionWithFallback(
+                chain,
+                route,
+                srcAddress,
+                isAffiliate
+            )
+        )
```
Comment (JaimeToca) [inline data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:131]: Do we care about all these error handling? or can we just do bodyOrThrow() and throw exception or map to exception with specific message?

File: data/src/main/kotlin/com/vultisig/wallet/data/api/models/quotes/KyberSwapQuoteJson.kt.kt (hunk 2 above)
Comment (aminsato) [reply data/src/main/kotlin/com/vultisig/wallet/data/api/models/quotes/KyberSwapQuoteJson.kt.kt:39]: Good point
Thanks

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt (hunk 5 above)
Comment (JaimeToca) [reply data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:155]: Probably we should just throw exception or error

Comment (aminsato) [reply data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:155]: The logic has been refactored to match the iOS codebase
https://github.com/vultisig/vultisig-ios/blob/4ac7e4519e65cdb932a4811a4c554abaa065bf88/VultisigApp/VultisigApp/Services/KyberSwap/KyberSwapService.swift#L82

File: data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt (hunk 9 above)
Comment (aminsato) [reply data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt:172]: I aligned the logic with the iOS version
https://github.com/vultisig/vultisig-ios/blob/4ac7e4519e65cdb932a4811a4c554abaa065bf88/VultisigApp/VultisigApp/Services/KyberSwap/KyberSwapService.swift#L109
